platforms/core-configuration/base-diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/ProjectReportTask.java

```diff
@@ -139,12 +138,10 @@ protected ProjectReportModel calculateReportModelFor(Project project) {
     }
 
     private List<ProjectReportModel> calculateChildrenProjectsFor(Project project) {
-        List<Project> childProjects = CollectionUtils.sort(ProjectHierarchyUtils.getChildProjectsForInternalUse(project));
-        List<ProjectReportModel> children = new ArrayList<>(childProjects.size());
-        for (Project childProject : childProjects) {
-            children.add(calculateReportModelFor(childProject));
-        }
-        return children;
+        return ((ProjectInternal) project).getOwner().getChildProjects().stream()
+            .sorted(ProjectOrderingUtil::compare)
```

Comment:
❔ Is sorting necessary, given that `getChildProjects` already returns a sorted list of children?

It's a bit concerning that the implementations of the two sortings seem disconnected. The util relies on depths + project path, while the project state relies on project identity paths (`Comparator.comparing(ProjectState::getIdentityPath)`). Are they equivalent? Can we keep only one of them?

Reply:
I took a look through all use of `getChildProjects` and didn't find any place that seemed to really care about the order. 

Here is a PR which moves all the usages to the method that does not do any sorting: https://github.com/gradle/gradle/pull/34936

Let's see what the test results say

Reply:
With very low priority, it might be nice to unify the sorting strategies. It seems _project_ sorting, as visible to the user, is also depth based, and that functionality is even exposed on the public interface, for no apparent reason.

https://github.com/gradle/gradle/blob/06653af6ac247e6118231af5bc06587dff5fe98b/subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java#L632-L640

They should probably be equivalent, but I wonder if there are edge cases